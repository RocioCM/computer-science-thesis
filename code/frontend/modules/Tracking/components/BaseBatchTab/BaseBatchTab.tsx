import React from 'react';
import { BaseBottleBatch } from '../../types';

interface Props {
  batch: BaseBottleBatch;
}

const BaseBatchTab: React.FC<Props> = ({ batch }) => {
  return (
    <div className="h-full w-full overflow-auto hide-scroll relative">
      {/* Información General */}
      <div className="grid grid-cols-2 gap-3xl">
        <div className="space-y-2">
          <h3 className="text-lg font-semibold border-b pb-2">
            Información General
          </h3>
          <p>
            <span className="font-semibold">Cantidad de envases:</span>{' '}
            {batch.quantity}
          </p>
          <p>
            <span className="font-semibold">Vendidos:</span>{' '}
            {batch.soldQuantity}
          </p>
          <p>
            <span className="font-semibold">Fecha de creación:</span>{' '}
            {new Date(batch.createdAt).toLocaleDateString()}
          </p>
          <p>
            <span className="font-semibold">Ubicación de origen:</span>{' '}
            {batch.bottleType.originLocation}
          </p>
        </div>
        {/* Detalles de la Botella */}
        <div>
          <h3 className="text-lg font-semibold border-b pb-2">
            Detalles de la Botella
          </h3>
          <div className="mt-2 space-y-2">
            <p>
              <span className="font-semibold">Peso:</span>{' '}
              {batch.bottleType.weight}g
            </p>
            <p>
              <span className="font-semibold">Color:</span>{' '}
              {batch.bottleType.color}
            </p>
            <p>
              <span className="font-semibold">Espesor:</span>{' '}
              {batch.bottleType.thickness} mm
            </p>
            <p>
              <span className="font-semibold">Forma:</span>{' '}
              {batch.bottleType.shapeType}
            </p>
          </div>
        </div>
      </div>

      {/* Composición */}
      <div className="mt-6">
        <h3 className="text-lg font-semibold border-b pb-2">Composición</h3>
        {batch.bottleType.composition.length > 0 ? (
          <table className="w-full mt-2 border border-gray-300 rounded-md">
            <thead>
              <tr className="bg-gray-100 text-gray-700">
                <th className="px-4 py-2 text-left">Material</th>
                <th className="px-4 py-2 text-left">Cantidad</th>
                <th className="px-4 py-2 text-left">Unidad</th>
              </tr>
            </thead>
            <tbody>
              {batch.bottleType.composition.map((material, index) => (
                <tr key={index} className="border-t">
                  <td className="px-4 py-2">{material.name}</td>
                  <td className="px-4 py-2">{material.amount}</td>
                  <td className="px-4 py-2">{material.measureUnit}</td>
                </tr>
              ))}
            </tbody>
          </table>
        ) : (
          <p className="text-gray-600 mt-2">
            No hay información de composición.
          </p>
        )}
      </div>
      {/* Información Adicional */}
      {batch.bottleType.extraInfo && (
        <div className="mt-6">
          <h3 className="text-lg font-semibold border-b pb-2">
            Información Adicional
          </h3>
          <p className="text-gray-700 mt-2">{batch.bottleType.extraInfo}</p>
        </div>
      )}
    </div>
  );
};

export default BaseBatchTab;
