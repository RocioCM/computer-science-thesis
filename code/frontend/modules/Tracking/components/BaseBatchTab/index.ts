import BaseBatchTab from './BaseBatchTab';

export default BaseBatchTab;
