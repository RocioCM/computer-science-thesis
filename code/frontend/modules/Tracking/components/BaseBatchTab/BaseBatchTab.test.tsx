import { render } from '@testing-library/react';
import BaseBatchTab from './BaseBatchTab';

describe('Tracking - BaseBatchTab', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  });

  afterEach(() => {
    const modalRoot = document.getElementById('modal-root');
    if (modalRoot) {
      document.body.removeChild(modalRoot);
    }
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <BaseBatchTab
        batch={{
          id: 1,
          bottleType: {
            color: 'Red',
            shapeType: 'Round',
            weight: 300,
            originLocation: 'Mza',
            thickness: 3,
            extraInfo: '900ml',
            composition: [{ name: 'Glass', amount: 100, measureUnit: '%' }],
          },
          owner: '0x1234567890abcdef1234567890abcdef12345678',
          quantity: 100,
          soldQuantity: 50,
          createdAt: new Date('2025-04-13T13:00:00').toISOString(),
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
