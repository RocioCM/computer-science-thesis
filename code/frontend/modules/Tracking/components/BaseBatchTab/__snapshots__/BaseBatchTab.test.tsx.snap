// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracking - BaseBatchTab Matches snapshot 1`] = `
<div>
  <div
    class="h-full w-full overflow-auto hide-scroll relative"
  >
    <div
      class="grid grid-cols-2 gap-3xl"
    >
      <div
        class="space-y-2"
      >
        <h3
          class="text-lg font-semibold border-b pb-2"
        >
          Información General
        </h3>
        <p>
          <span
            class="font-semibold"
          >
            Cantidad de envases:
          </span>
           
          100
        </p>
        <p>
          <span
            class="font-semibold"
          >
            Vendidos:
          </span>
           
          50
        </p>
        <p>
          <span
            class="font-semibold"
          >
            Fecha de creación:
          </span>
           
          30/4/2025
        </p>
        <p>
          <span
            class="font-semibold"
          >
            Ubicación de origen:
          </span>
           
          Mza
        </p>
      </div>
      <div>
        <h3
          class="text-lg font-semibold border-b pb-2"
        >
          Detalles de la Botella
        </h3>
        <div
          class="mt-2 space-y-2"
        >
          <p>
            <span
              class="font-semibold"
            >
              Peso:
            </span>
             
            300
            g
          </p>
          <p>
            <span
              class="font-semibold"
            >
              Color:
            </span>
             
            Red
          </p>
          <p>
            <span
              class="font-semibold"
            >
              Espesor:
            </span>
             
            3
             mm
          </p>
          <p>
            <span
              class="font-semibold"
            >
              Forma:
            </span>
             
            Round
          </p>
        </div>
      </div>
    </div>
    <div
      class="mt-6"
    >
      <h3
        class="text-lg font-semibold border-b pb-2"
      >
        Composición
      </h3>
      <table
        class="w-full mt-2 border border-gray-300 rounded-md"
      >
        <thead>
          <tr
            class="bg-gray-100 text-gray-700"
          >
            <th
              class="px-4 py-2 text-left"
            >
              Material
            </th>
            <th
              class="px-4 py-2 text-left"
            >
              Cantidad
            </th>
            <th
              class="px-4 py-2 text-left"
            >
              Unidad
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="border-t"
          >
            <td
              class="px-4 py-2"
            >
              Glass
            </td>
            <td
              class="px-4 py-2"
            >
              100
            </td>
            <td
              class="px-4 py-2"
            >
              %
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="mt-6"
    >
      <h3
        class="text-lg font-semibold border-b pb-2"
      >
        Información Adicional
      </h3>
      <p
        class="text-gray-700 mt-2"
      >
        900ml
      </p>
    </div>
  </div>
</div>
`;
